base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java

```diff
@@ -131,16 +140,18 @@ private void writeLanguageInfo(
     }
   }
 
-  private static @NotNull Map<String, String> getLanguageStringMap(BlazeProjectDataManager manager, Project project) {
-    var projectData = Optional.ofNullable(manager.getBlazeProjectData()); // It can be empty on intial sync. Fall back to no lauguage support
-    var blazeProjectData = BlazeProjectDataManager.getInstance(project).getBlazeProjectData();
+  private static @NotNull Map<String, String> getLanguageStringMap(BlazeProjectDataManager manager) {
+    var projectData = Optional.ofNullable(manager.getBlazeProjectData()); // It can be empty on intial sync. Fall back to no language support
     var activeLanguages = projectData.map(it -> it.getWorkspaceLanguageSettings().getActiveLanguages()).orElse(ImmutableSet.of());
-    var isJavaEnabled = activeLanguages.contains(LanguageClass.JAVA)
-            && blazeProjectData != null
-            && blazeProjectData.getExternalWorkspaceData().getByRepoName("rules_java") != null;
-    var isPythonEnabled = activeLanguages.contains(LanguageClass.PYTHON)
-            && blazeProjectData != null
-            && blazeProjectData.getExternalWorkspaceData().getByRepoName("rules_python") != null;
+//    var externalWorkspaceData = projectData.map(BlazeProjectData::getExternalWorkspaceData).orElse(null);
+    var isJavaEnabled = activeLanguages.contains(LanguageClass.JAVA);
```

Comment:
how about keeping it for legacy sync, and disabling the check for query sync?

Reply:
so instead of commenting out, we could add `|| isQuerySync`

Reply:
sure, will do.